src/app/pages/properties/add-property/add-property.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, OnInit } from "@angular/core";
+import {
+  FormGroup,
+  FormControl,
+  FormBuilder,
+  Validators
+} from "@angular/forms";
+import { PropertyService } from "src/app/services/property/property.service";
+
+@Component({
+  selector: "app-add-property",
+  templateUrl: "./add-property.component.html",
+  styleUrls: ["./add-property.component.scss"]
+})
+export class AddPropertyComponent implements OnInit {
+  addPropertyForm: FormGroup;
+
+  constructor(
+    private fb: FormBuilder,
+    private propertyService: PropertyService
+  ) {}
+
+  numberLengthLess(control: FormControl): { [key: string]: boolean } | null {
+    if (control.value.toString().length > 14) {
+      return { maxLen: true };
+    }
+
+    return null;
+  }
+
+  ngOnInit() {
+    this.addPropertyForm = this.fb.group({
+      title: ["", [Validators.required, Validators.minLength(5)]],
+      description: ["", [Validators.required, Validators.minLength(2)]],
+      price: ["", [Validators.required, this.numberLengthLess]],
+      lot_size: ["", [Validators.required, this.numberLengthLess]],
+      property_type: ["", [Validators.required, Validators.maxLength(1)]],
+      purchase_plan: ["", [Validators.required, Validators.maxLength(1)]],
+      state: ["", [Validators.required, Validators.minLength(3)]],
+      street: ["", [Validators.required, Validators.minLength(3)]],
+      city: ["", [Validators.required, Validators.minLength(3)]],
+      latitude: ["", [Validators.required]],
+      longitude: ["", [Validators.required]],
+      image_main: ["", [Validators.required]],
+      video: [""],
+      image_others: [''],
+      bedrooms: [""],
+      bathrooms: [""],
+      garages: [""]
+    });
+  }
+
+  addNewProperty(value: any) {
```

Comment:
Also, since we are adding new property and using typescript, wouldn't it be better to specify the actual type of the value that the `addNewProperty` method takes as a parameter instead of any?